app/src/main/java/com/google/samples/apps/sunflower/GardenFragment.kt

```diff
@@ -101,23 +115,36 @@ class GardenFragment : Fragment() {
     private class GardenSelectionObserver(
         private val selectionTracker: SelectionTracker<Long>,
         private val onSelectionChangedListener: (Int) -> Unit
-    ) :
-        SelectionTracker.SelectionObserver<Long>(),
-        OnBackPressedCallback {
+    ) : SelectionTracker.SelectionObserver<Long>() {
 
         override fun onSelectionChanged() {
             super.onSelectionChanged()
             onSelectionChangedListener(selectionTracker.selection?.size() ?: 0)
         }
+    }
 
-        // If the back button is pressed and there's a selection, clear the selection, rather
-        // than potentially exiting the app.
-        override fun handleOnBackPressed() =
-            if (selectionTracker.hasSelection()) {
-                selectionTracker.clearSelection()
-                true
-            } else {
-                false
-            }
+    /**
+     * Inner class to encapsulate handling of the [ActionMode] in my garden.
+     */
+    private inner class GardenActionModeCallback : ActionMode.Callback {
```

Comment:
Take care, `inner class` has an implicit pointer to parent class(Fragment).

I do not know that Google uses for a standard solution to face the `inner class` (like java inner class), however, in my recent project, I did like this, which could be a bit of advice here:

```kotlin

inner class Foo(lifecycleOwner: LifecycleOwner) : LifecycleObserver {
        private var lifecycleOwner: LifecycleOwner? = lifecycleOwner
  
        @OnLifecycleEvent(Lifecycle.Event.ON_DESTROY)
        fun onDestroy() {
            lifecycleOwner?.lifecycle?.removeObserver(this)
            lifecycleOwner = null
        }

        fun foo() = lifecycleOwner?.let {
                //TODO with foo task here.
         }
}

```

Reply:
Sure, but the `inner class` isn't passed anywhere, and so, while the fragment and inner class point at one another, nothing will prevent them both from being freed when the Fragment is released.